File: Escola/Cursos.Site/app/services/LocaleventoService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 10/22/2017 8:08:43 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("LocaleventoService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Localevento/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Localevento/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Localevento, callback) {
            $http.post(serviceBase + "Localevento/post", Localevento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Localevento, callback) {
            $http.post(serviceBase + "Localevento/put", Localevento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Localevento, callback) {
            $http.post(serviceBase + "Localevento/delete", Localevento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/LocaleventoService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  LocaleventoService
      /// </summary>
      public partial class LocaleventoService : BaseServices, ILocaleventoService
      {

		public LocaleventoService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_LOCALEVENTO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_LOCALEVENTO> entity)
        {
            UnitOfWork.LocaleventoRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_LOCALEVENTO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_LOCALEVENTO entity)
        {	
            UnitOfWork.LocaleventoRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_LOCALEVENTO
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_LOCALEVENTO entity)
        {
            UnitOfWork.LocaleventoRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_LOCALEVENTO
        /// </summary>
        /// <returns>IEnumerable EV_LOCALEVENTO</returns>
        public IEnumerable<EV_LOCALEVENTO> GetAll()
        {
			return UnitOfWork.LocaleventoRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_LOCALEVENTO
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_LOCALEVENTO entity)
        {
            UnitOfWork.LocaleventoRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_LOCALEVENTO By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_LOCALEVENTO</returns>
		public EV_LOCALEVENTO GetById(int id)
        {
			return UnitOfWork.LocaleventoRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_LOCALEVENTO> FindBy(Expression<Func<EV_LOCALEVENTO, bool>> predicate)
        {
            return UnitOfWork.LocaleventoRepository.FindBy(predicate);
        }
      }
}
